File: backend/internal/model/ai.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type AIProvider struct {
	ID               string         `gorm:"primaryKey;size:36" json:"id"`
	UserID           uint           `gorm:"not null;index" json:"user_id"`
	Name             string         `gorm:"size:100;not null" json:"name"`
	ProviderType     string         `gorm:"size:50;not null;default:openai_compatible" json:"provider_type"`
	BaseURL          string         `gorm:"size:500;not null" json:"base_url"`
	APIKeyCiphertext string         `gorm:"type:text" json:"-"`
	Model            string         `gorm:"size:100;not null" json:"model"`
	Enabled          bool           `gorm:"default:true" json:"enabled"`
	CreatedAt        time.Time      `json:"created_at"`
	UpdatedAt        time.Time      `json:"updated_at"`
	DeletedAt        gorm.DeletedAt `gorm:"index" json:"-"`
}

type AIReport struct {
	ID            string         `gorm:"primaryKey;size:36" json:"id"`
	UserID        uint           `gorm:"not null;index" json:"user_id"`
	ReportType    string         `gorm:"size:30;not null;index" json:"report_type"`
	PeriodStart   time.Time      `gorm:"not null;index" json:"period_start"`
	PeriodEnd     time.Time      `gorm:"not null;index" json:"period_end"`
	Status        string         `gorm:"size:30;not null;default:pending" json:"status"`
	SnapshotJSON  string         `gorm:"type:text" json:"snapshot_json"`
	ContentJSON   string         `gorm:"type:text" json:"content_json"`
	ProviderID    string         `gorm:"size:36;index" json:"provider_id"`
	ProviderName  string         `gorm:"size:100" json:"provider_name"`
	Model         string         `gorm:"size:100" json:"model"`
	PromptVersion string         `gorm:"size:30" json:"prompt_version"`
	ErrorMessage  string         `gorm:"type:text" json:"error_message,omitempty"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `gorm:"index" json:"-"`
}

File: web/src/api/ai.ts
import { get, post, put, del } from '@/utils/request'

export interface AIProvider {
  id: string
  name: string
  provider_type: string
  base_url: string
  model: string
  enabled: boolean
  created_at: string
  updated_at: string
}

export interface AIProviderPreset {
  id: string
  name: string
  provider_type: string
  base_url: string
  model: string
  models: string[]
}

export interface SaveAIProviderParams {
  name: string
  provider_type?: string
  base_url: string
  api_key?: string
  model: string
  enabled: boolean
}

export interface AIReport {
  id: string
  report_type: string
  period_start: string
  period_end: string
  status: string
  snapshot_json: string
  content_json: string
  provider_id: string
  provider_name: string
  model: string
  prompt_version: string
  error_message?: string
  created_at: string
  updated_at: string
}

export interface GenerateAIReportParams {
  report_type: string
  provider_id?: string
  period_start: string
  period_end: string
}

export const aiApi = {
  listProviderPresets(): Promise<AIProviderPreset[]> {
    return get<AIProviderPreset[]>('/ai/providers/presets')
  },

  listProviders(): Promise<AIProvider[]> {
    return get<AIProvider[]>('/ai/providers')
  },

  createProvider(params: SaveAIProviderParams): Promise<AIProvider> {
    return post<AIProvider>('/ai/providers', params)
  },

  updateProvider(id: string, params: SaveAIProviderParams): Promise<AIProvider> {
    return put<AIProvider>(`/ai/providers/${id}`, params)
  },

  deleteProvider(id: string): Promise<void> {
    return del<void>(`/ai/providers/${id}`)
  },

  testProvider(id: string): Promise<{ ok: boolean }> {
    return post<{ ok: boolean }>(`/ai/providers/${id}/test`)
  },

  listReports(): Promise<AIReport[]> {
    return get<AIReport[]>('/ai/reports')
  },

  getReport(id: string): Promise<AIReport> {
    return get<AIReport>(`/ai/reports/${id}`)
  },

  generateReport(params: GenerateAIReportParams): Promise<AIReport> {
    return post<AIReport>('/ai/reports/generate', params)
  },

  deleteReport(id: string): Promise<void> {
    return del<void>(`/ai/reports/${id}`)
  }
}
